unittests/test_warn_missing_include_dirs.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2014-2017 Insight Software Consortium.
+# Copyright 2004-2009 Roman Yakovenko.
+# Distributed under the Boost Software License, Version 1.0.
+# See http://www.boost.org/LICENSE_1_0.txt
+
+from contextlib import redirect_stdout
+import io
+import sys
+import os
+import unittest
+
+sys.path.insert(1, os.path.join(os.curdir, '..'))
+sys.path.insert(1, "../pygccxml")
+
+from pygccxml import parser  # nopep8
+from pygccxml import utils  # nopep8
+
+
+class Test(unittest.TestCase):
+
+    def test_config(self):
+        """
+            Test that a missing include directory is printing a warning,
+            not raising an error
+        """
+
+        # Some code to parse for the example
+        code = "int a;"
+
+        # Find the location of the xml generator (castxml or gccxml)
+        generator_path, name = utils.find_xml_generator()
+
+        f = io.StringIO()
+        # Path given as include director doesn't exist
+        config = parser.xml_generator_configuration_t(
```

Comment:
Per f2f, can we include a mechanism to pass a `-Werror` proxy, so that this warning is still an error?

Context:
In https://github.com/RobotLocomotion/drake/pull/14265, we actually do want to ensure those directories exist (I think).

Reply:
OK I think this was resolved by a separate mechanism. I dunno how to resolve my own review comment, though.